### PR TITLE
fix the log when data-path is not exist

### PR DESCRIPTION
**Pull request by mdh67899**
https://github.com/nsqio/nsq/issues/1217

### CODE REVIEW

File: nsqd/nsqd.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func New(opts *Options) (*NSQD, error) {
 
 	err = n.dl.Lock()
 	if err != nil {
-		return nil, fmt.Errorf("--data-path=%s in use (possibly by another instance of nsqd)", dataPath)
+		return nil, fmt.Errorf("trying lock --data-path=%s failed: %v (possibly by another instance of nsqd)", dataPath, err)
```
Comment (jehiah) [inline nsqd/nsqd.go]: @mdh67899 what is the specific error you are getting? If you can share that it might help to better understand how to structure this error to handle the normal case (Another nsqd is running) and the case it sounds like you encountered (filesystem support)

Comment (mdh67899) [reply nsqd/nsqd.go]: @jehiah If the directory specified by ```--data-path``` does not exist, then nsqd startup will fail when trying to execute ```n.dl.Lock ()```.

In addition, if there are other nsqd processes locking the ```--data-path``` , an error will also be reported
https://github.com/nsqio/nsq/blob/5b67f580f3cbb63c02c4678e76896f5807147145/internal/dirlock/dirlock.go#L22-L33

If the ```n.dl.Lock ()``` operation fails, the error should be printed in the log for troubleshooting.
or do you think it is a good idea to create data path when ```--data-path``` is not exist?

Comment (mdh67899) [reply nsqd/nsqd.go]: @jehiah in issue #1217, it's not a filesystem support problem, the reason is the ```--data-path=/data/nsq``` is no exist in docker container

Comment (ploxiln) [reply nsqd/nsqd.go]: I think it is a good idea to add the underlying error text here. I think it could be worded in a simpler way, maybe just `"failed to lock data-path: %v"`, and we may not need to include the path explicitly, nor the "(possibly by another instance ..." part . The main error possibilities are (according to my testing just now with this branch and go-1.13 on macOS):
  * "cannot flock directory ./tmp - resource temporarily unavailable"
  * "open ./tmp2: no such file or directory"
  * "open ./tmp2: permission denied"

We could add the "possibly by another instance" explanation just for the flock error ...

Comment (jehiah) [reply nsqd/nsqd.go]: thanks for the context @ploxiln. I think it would be good to keep explicit guidance in the flock error case that nsqd doesn't support multiple instances running w/ the same data directory.

Comment (mdh67899) [reply nsqd/nsqd.go]: @jehiah @ploxiln Do you think it is appropriate to adjust the code this way?

nsq/internal/dirlock/dirlock.go
```
 func (l *DirLock) Lock() error { 
 	f, err := os.Open(l.dir) 
 	if err != nil { 
 		return fmt.Errorf("cannot open directory %s - %s", l.dir, err) 
 	} 
 	l.f = f 
 	err = syscall.Flock(int(f.Fd()), syscall.LOCK_EX|syscall.LOCK_NB) 
 	if err != nil { 
 		return fmt.Errorf("cannot flock directory %s - %s (possibly by another instance of nsqd)", l.dir, err)
  	} 
 	return nil 
 } 
```

nsqd/nsqd.go

```
	err = n.dl.Lock()
	if err != nil {
		return nil, fmt.Errorf("failed to lock data-path: %v", err)
	}
```

Comment (ploxiln) [reply nsqd/nsqd.go]: Yeah, adding the "possibly by another instance" to the Flock error return works for me. With the other changes, I would rephrase it "(possibly in use by another instance of nsqd)" here.

I don't think the Open error needs any additions, the text seems to already include "open {dir}".

I tested briefly in a linux VM and saw the same error text from Open and Flock.

**Comment by mdh67899** [comment]
@ploxiln PTAL
